sound/soc/intel/boards/sof_ssp_common.h

```diff
@@ -65,7 +65,29 @@ enum sof_ssp_codec {
 	CODEC_RT1308,
 };
 
+/*
+ * sof_ssp_common: common data for machine drivers
+ *
+ * @hdmi_pcm_list: sof_hdmi_pcm list to support Intel HDMI links
+ * @private_data: machine driver private data
+ * @common_hdmi_codec_drv: 1 to use snd-hda-codec-hdmi driver; 0 to use hdac-hdmi
+ * @idisp_codec: 1 to indicate idisp codec is present
+ */
+struct sof_ssp_common {
+	struct list_head hdmi_pcm_list;
+	void *private_data;
+
+	unsigned long common_hdmi_codec_drv:1;
+	unsigned long idisp_codec:1;
+};
+
```

Comment:
good idea but why not move this to a different header that's not related to SSP at all? that way we could also use the same helpers for SoundWire devices.

Reply:
an new header sof_sdw_ssp_hdmi.h is created for this